File: app/channels/sync_channel.rb
class SyncChannel < ApplicationCable::Channel
  def subscribed
    stream_from "updates"
  end

  def unsubscribed
  end

  def set_chart_ticker(data)
    Setting.replace_chart_ticker(data['ticker'])
    SyncChannel.push :reload_chart
  end

  class << self
    def push(reason = 'updated')
      ActionCable.server.broadcast 'updates', { reason: reason }
    end
  end
end

__END__
SyncChannel.push 'prices'

File: app/javascript/channels/sync_channel.js
import consumer from "./consumer"

const syncChannel = consumer.subscriptions.create("SyncChannel", {
  connected() {
  },

  disconnected() {
  },

  received(data) {
    console.log('SyncChannel received data', data)
    if (data.reason == 'prices' && window.DashboardPeriod == '') {
      location.reload()
    } else if (data.reason == 'candles') {
      document.dispatchEvent(new Event('chart-reload-data'))
    } else if (data.reason == 'reload_chart') {
      if ($qs('.charts-page')) {
        location.reload()
      }
    }
  },

  setChartTicker(ticker) {
    this.perform("set_chart_ticker", { ticker: ticker })
  },
});

window.syncChannel = syncChannel
